lib/dtutils.lua

```diff
@@ -0,0 +1,866 @@
+  local ret
+  if file ~= nil then 
+    io.close(file) 
+    dt.print_error("true checkIfFileExists: "..filepath)
+    ret = true
+  else 
+    dt.print_error(filepath.." not found")
+    ret = false
+  end
+  return ret
+end
+
+--[[
+  NAME
+    dutils.filename_increment - add a two digit increment to a filename
+
+  SYNOPSIS
+    result = dtutils.filename_increment(filepath)
+      filepath - file to increment
+
+  DESCRIPTION
+    filename_increment solves the problem of filename confllict by adding an 
+    increment to the filename.  If the supplied filename has no increment then 
+    "01" is added to the basename.  If the filename already has an increment, then
+    1 is added to it and the filename returned.
+
+  RETURN VALUE
+    result - the incremented filename
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.filename_increment(filepath)
+
+  -- break up the filepath into parts
+  local path = dtutils.get_path(filepath)
+  local basename = dtutils.get_basename(filepath)
+  local filetype = dtutils.get_filetype(filepath)
+
+  -- check to see if we've incremented before
+  local increment = string.match(basename, "_(%d-)$")
+
+  if increment then
+    -- we do 2 digit increments so make sure we didn't grab part of the filename
+    if string.len(increment) > 2 then
+      -- we got the filename so set the increment to 01
+      increment = "01"
+    else
+      increment = string.format("%02d", tonumber(increment) + 1)
+      basename = string.gsub(basename, "_(%d-)$", "")
+    end
+  else
+    increment = "01"
+  end
+  local incremented_filepath = path .. basename .. "_" .. increment .. "." .. filetype
+
+  dt.print_error("original file was " .. filepath)
+  dt.print_error("incremented file is " .. incremented_filepath)
+
+  return incremented_filepath
+end
+
+--[[
+  NAME
+    dutils.groupIfNotMember - group a new file with an existing one
+
+  SYNOPSIS
+    result = dtutils.groupIfNotMember(image, new_image)
+      image - an existing image in the darktable database
+      new_image - the new image to group with the existing image
+
+  DESCRIPTION
+    groupIfNotMember checks a file to see if it's a member of the group
+    with an existing image.  If not, then the group leader of the existing 
+    image is found and the new image is grouped with it.
+
+  RETURN VALUE
+    none
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.groupIfNotMember(img, new_img)
+  local image_table = img:get_group_members()
+  local is_member = false
+  for _,image in ipairs(image_table) do
+    dt.print_error(image.filename .. " is a member")
+    if image.filename == new_img.filename then
+      is_member = true
+      dt.print_error("Already in group")
+    end
+  end
+  if not is_member then
+    dt.print_error("group leader is "..img.group_leader.filename)
+    new_img:group_with(img.group_leader)
+    dt.print_error("Added to group")
+  end
+end
+
+--[[
+  NAME
+    dutils.sanitize_filename - take care of spaces in filenames
```

Comment:
I wrote this when I was really new to lua, before I became "enlightened" ;).  This should probably go away since there are better ways to do this